context_test.go

```diff
@@ -214,6 +214,67 @@ func (*ContextSuite) TestApplyConfigAdditive(c *gc.C) {
 		})
 }
 
+func (*ContextSuite) TestApplyConfigLabels(c *gc.C) {
+	context := loggo.NewContext(loggo.WARNING)
+	context.GetLogger("a.b", "ONE")
+	context.GetLogger("c.d", "ONE")
+	context.GetLogger("e", "TWO")
+
+	context.ApplyConfig(loggo.Config{"[ONE]": loggo.TRACE})
+	context.ApplyConfig(loggo.Config{"[TWO]": loggo.DEBUG})
+
+	c.Assert(context.Config(), gc.DeepEquals,
+		loggo.Config{
+			"":    loggo.WARNING,
+			"a.b": loggo.TRACE,
+			"c.d": loggo.TRACE,
+			"e":   loggo.DEBUG,
+		})
+	c.Assert(context.CompleteConfig(), gc.DeepEquals,
+		loggo.Config{
+			"":    loggo.WARNING,
+			"a":   loggo.UNSPECIFIED,
+			"a.b": loggo.TRACE,
+			"c":   loggo.UNSPECIFIED,
+			"c.d": loggo.TRACE,
+			"e":   loggo.DEBUG,
+		})
+}
+
+func (*ContextSuite) TestApplyConfigLabelsAddative(c *gc.C) {
+	context := loggo.NewContext(loggo.WARNING)
+	context.ApplyConfig(loggo.Config{"[ONE]": loggo.TRACE})
```

Comment:
I think there could be a lot of potential syntax for labels. Is the idea that you might do:

`logging-config="<root>=INFO;[http,upgrades]=DEBUG"`

or that you would do:
`logging-config="<root>=INFO;[http]=DEBUG,[upgrades]=DEBUG"`

FWIW, I would tend to make *labels* be lowercase, to align with how we do modules/namespaces.

I'm not a big fan of `[label]=` vs `namespace=`, because the former really looks like a list, so `[label1,label2]=`.

You could use all sorts of potential syntax/meta characters. Like "<root>=INFO;^label=;#label=;.."
I actually kind of like # as a label, as hashtag is the way that you do labels in other systems.
It tends to be a shell comment character, but so is pretty much any special character, so you already have to quote because of <root>

FWIW, I'd also like us to support:
`logging-config="DEBUG;h=INFO;b.b=TRACE"`
so that you don't have to specify <root> which I find particularly clumsy.

Reply:
> `logging-config="DEBUG;h=INFO;b.b=TRACE"`

That's short hand for `<root>=DEBUG;h=INFO;b.b=TRACE` right?